### ISSUE TITLE
Live Show - Gameplay

### ISSUE THREAD

**Issue by sr229** [ClarityMoe/Lovewing#29]
Here's the design for Gameplay

### Group Stage

![](https://i.imgur.com/tYGTgXp.png)

### Solo Stage 

![image](https://user-images.githubusercontent.com/14976516/34913924-b7104004-f943-11e7-9763-7b6af24ea17c.png)

#### Notes
- Fonts here are [Venera 700](https://www.dafontfree.net/freefonts-venera-f161371.htm). It's possible to add glow in osu!Framework-side.
- Middle Circle is the origin of the notes and the player's HP bar. Consider the following colors:
      - Material Green if HP is 50%
      - Material Orange to Red if less than 20% (Critical)
      - Material Blue if full HP
- Live Mode (The viewer count and the live indicator) only appears if the player is being spectated.

**Comment by sr229** [comment]
Reminder to keep it #14-compliant

**Comment by MikuAuahDark** [comment]
I'll just gonna comment here about this live show UI design, because I can't implement all of the UI design correctly in livesim2. Probably you can use this as motivation, but maybe not.

* "The middle circle" (whatever the fuck you call it, reference from #26) is the note spawning position. The actual SIF game spawns the note in 480x160 (assume 960x640 resolution, because that's how SIF handles it), and if you use speed 6 (1000ms) in SIF, it will took 1 second to let the note travel **exactly** [400 pixels](https://cdn.discordapp.com/attachments/207972536393138177/427732177736826881/sif_note_explanation.png). I've written [article](https://auahdark687291.blogspot.co.id/2018/03/sif-note-handling.html) about how SIF note timing is implemented. There's green text that writes "Distance to note spawn position. No matter which idol icon you choose, the distance always equal to 400 pixels". But with this UI, the distance is varies. When I put something in SIF note spawning position, then it will spawn [above the "middle circle"](https://cdn.discordapp.com/attachments/341943871443304450/438279061677146112/temporary.png). This will make the feel that the note targetting center position will be slow (<350px/second), but notes in leftmost or rightmost will be very fast (>470px/second). If you want exact number, do a phytagoras distance calculation between the "middle circle" and all of the tap icons.

* Blur is nice, but don't overuse it. I know this thing won't be available in mobile device, but if it is, I doubt it will run fast in mid-end phones. It may run fast enough (or acceptable) in my ZenFone 2 because it's equipped with Intel CPU and iOS GPU, but it will be real slow in my Mi A1. Performance optimization aside, don't do the blur in real-time, but cache the blur result and re-blur if necessary.

* Back to middle circle thing. "The middle circle (whatever the fuck you call it) should bounce relative to the song's BPM". That would be, uh, requires complex computation time (I'm talking about FFT here). Not to mention the blur that already I mentioned before. However there's a to prevent the whole FFT computation entirely. Either by forcing beatmap to add "beat" points, or by forcing beatmap to have "BPM" field.

* I'm not sure what does that "990" mean. I would assume that's the "Notes remaining" or "base score add/tap". Also the "LIVE" text is somehow unnecessary, because this is live part, unless that "LIVE" text will change on certain circumstances, then I don't see a reason it's there.

* Tap icon placement should comply with SIF, or people find it's weird. I know this is redesign (and I think it doesn't have to follow SIF tap icon placement), but I think it should comply about this one. The tap icon placement in SIF is actually clever. As explained in the first point, SIF tap icon placement will make note travel at constant speed (400px/second). If you don't believe this, you can take [these coordinates](https://github.com/MikuAuahDark/livesim2/blob/master/livesim.lua#L42), add by 64 on both axis, and calculate the distance to 480x160 using phytagoras distance formula.

[This is how I implement that UI in livesim2](https://cdn.discordapp.com/attachments/341943871443304450/438287345234477066/screenshot_2018_04_24_18_36_42_987.png). Doesn't look like what you expected, but I try to be SIF-compilant as possible. I moved the "middle circle" position to be in 480x160 and the combo counter just below it.

If you need to discuss something further, in a fast way, you can hit me up in Discord.

**Comment by jorolf** [comment]
Can you join our discord? https://discord.gg/rmMTZue

**Comment by MikuAuahDark** [comment]
Yes, no problem. Just hit me up (AuahDark#4586) if you need to talk.

**Comment by MikuAuahDark** [comment]
Man for real better show up.

**Comment by sr229** [comment]
@MikuAuahDark the design is spiritual to the SIF layout, and the dev team has he freedom to iterate out the original concept. And for UX reasons, the note spawn has been reused to be the health indicator as well. It bounces with the beat and we could make it compliant layot wise.

As for HC alignment, I tried to make it close as possible but for UX reasons again, I tried to conform it on widescreens.

And the blur is for concept reasons, not really part of the Gameplay UI.

**Comment by MikuAuahDark** [comment]
I would say, adapt for 2:3 screen ratio (960x640 or multiple of it) and do
something similar to letterbox (but not discarding draw area outside it).
That's how SIF implements full screen. There are tablet that even runs in
4:3 ratio if you were wondering.

Pada tanggal 25 Apr 2018 8.52 AM, "Capuccino" <notifications@github.com>
menulis:

@MikuAuahDark <https://github.com/MikuAuahDark> the design is spiritual to
the SIF layout, and the dev team has he freedom to iterate out the original
concept. And for UX reasons, the note spawn has been reused to be the
health indicator as well. It bounces with the beat and we could make it
compliant layot wise.

As for HC alignment, I tried to make it close as possible but for UX
reasons again, I tried to conform it on widescreens.

And the blur is for concept reasons, not really part of the Gameplay UI.

—
You are receiving this because you were mentioned.

Reply to this email directly, view it on GitHub
<https://github.com/ProjectLovewing/Lovewing/issues/29#issuecomment-384125628>,
or mute the thread

**Comment by sr229** [comment]
just to add, we're implementing spectator mode, that's the reason why the live indicator and the vkewer count indicator exists @MikuAuahDark
